From 2bbe4ab81fd6476beba56bf9eb6b69bf94e2e2cf Mon Sep 17 00:00:00 2001
From: Tomeu Vizoso <tomeu.vizoso@collabora.com>
Date: Thu, 30 Apr 2020 10:01:13 +0200
Subject: [PATCH] panfrost: Have format.block be MALI_BLOCK_TILED on Bifrost

We don't know yet why this is needed on Bifrost.

Signed-off-by: Tomeu Vizoso <tomeu.vizoso@collabora.com>
---
 src/gallium/drivers/panfrost/pan_mfbd.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/gallium/drivers/panfrost/pan_mfbd.c b/src/gallium/drivers/panfrost/pan_mfbd.c
index 1350e17485a..219658a8a35 100644
--- a/src/gallium/drivers/panfrost/pan_mfbd.c
+++ b/src/gallium/drivers/panfrost/pan_mfbd.c
@@ -199,6 +199,7 @@ panfrost_mfbd_set_cbuf(
         struct pipe_surface *surf)
 {
         struct panfrost_resource *rsrc = pan_resource(surf->texture);
+        struct panfrost_device *dev = pan_device(surf->context->screen);
 
         unsigned level = surf->u.tex.level;
         unsigned first_layer = surf->u.tex.first_layer;
@@ -212,7 +213,10 @@ panfrost_mfbd_set_cbuf(
         /* Now, we set the layout specific pieces */
 
         if (rsrc->layout == MALI_TEXTURE_LINEAR) {
-                rt->format.block = MALI_BLOCK_LINEAR;
+                if (dev->quirks & IS_BIFROST)
+                        rt->format.block = MALI_BLOCK_TILED; /* TODO: Why? */
+                else
+                        rt->format.block = MALI_BLOCK_LINEAR;
                 rt->framebuffer = base;
                 rt->framebuffer_stride = stride / 16;
         } else if (rsrc->layout == MALI_TEXTURE_TILED) {
-- 
2.26.2

